contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +157,230 @@ contract OETHVaultCore is VaultCore {
+     */
+    function requestWithdrawal(uint256 _amount)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 requestId, uint256 queued)
+    {
+        // Burn the user's OETH
+        // This also checks the requester has enough OETH to burn
+        oUSD.burn(msg.sender, _amount);
+
+        requestId = withdrawalQueueMetadata.nextWithdrawalIndex;
+        queued = withdrawalQueueMetadata.queued + _amount;
+
+        // store the next withdrawal request
+        withdrawalQueueMetadata.nextWithdrawalIndex = uint128(requestId + 1);
+        withdrawalQueueMetadata.queued = uint128(queued);
+
+        emit WithdrawalRequested(msg.sender, requestId, _amount, queued);
+
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            timestamp: uint40(block.timestamp),
+            amount: uint128(_amount),
+            queued: uint128(queued)
+        });
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawal once it is claimable.
+     * This request can be claimed once the withdrawal queue's `claimable` amount
+     * is greater than or equal this request's `queued` amount.
+     * There is no minimum time or block number before a request can be claimed. It just needs
+     * enough WETH liquidity in the Vault to satisfy all the outstanding requests to that point in the queue.
+     * OETH is converted to WETH at 1:1.
+     * @param _requestId Unique ID for the withdrawal request
+     * @return amount Amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawal(uint256 _requestId)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 amount)
+    {
+        amount = _claimWithdrawal(_requestId);
+
+        // transfer WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawals once they are claimable.
+     * This requests can be claimed once the withdrawal queue's `claimable` amount
+     * is greater than or equal each request's `queued` amount.
+     * If one of the requests is not claimable, the whole transaction will revert with `queue pending liquidity`.
+     * @param _requestIds Unique ID of each withdrawal request
+     * @return amounts Amount of WETH received for each request
+     * @return totalAmount Total amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawals(uint256[] memory _requestIds)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256[] memory amounts, uint256 totalAmount)
+    {
+        amounts = new uint256[](_requestIds.length);
+        for (uint256 i = 0; i < _requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(_requestIds[i]);
+            totalAmount += amounts[i];
```

Comment:
nit: `withdrawalQueueMetadata` is updated multiple times when claiming several withdrawal requests, Could probably save some gas by having a separate method for multiple claims instead of just  looping through the same method several times